File: in_Python/0202 Happy Number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        seen = set()       
        while n not in seen:
            seen.add(n)
            res = 0
            s = str(n)
            for i in list(s):
                res += int(i) ** 2
            n = res
        return n == 1
        

File: in_Java/0202 Happy Number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> hs = new HashSet<>();
        int t;
        while(true){
            t = 0;
            while(n != 0){
                t += (n % 10) * (n % 10);
                n /= 10;
            }
            if(hs.contains(t)) return false;
            else if(t == 1) return true;
            else hs.add(t);
            n = t;
        }
    }
}